tests/subsys/event_manager/src/events/data_event.c

```diff
@@ -13,11 +13,11 @@ static void profile_data_event(struct log_event_buf *buf,
 	struct data_event *event = cast_data_event(eh);
 
 	ARG_UNUSED(event);
-	profiler_log_encode_u32(buf, event->val1);
-	profiler_log_encode_u32(buf, event->val2);
+	profiler_log_encode_u8(buf, event->val1);
+	profiler_log_encode_u16(buf, event->val2);
 	profiler_log_encode_u32(buf, event->val3);
-	profiler_log_encode_u32(buf, event->val1u);
-	profiler_log_encode_u32(buf, event->val2u);
+	profiler_log_encode_u8(buf, event->val1u);
+	profiler_log_encode_u16(buf, event->val2u);
 	profiler_log_encode_u32(buf, event->val3u);
```

Comment:
Consider adding profiling for string data too. It could be good to cover all of the supported datatypes here.